providers/aws/resources/aws_vpc.go

```diff
@@ -101,6 +101,50 @@ func (a *mqlAws) getVpcs(conn *connection.AwsConnection) []*jobpool.Job {
 	return tasks
 }
 
+func (a *mqlAwsVpc) endpoints() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	vpc := a.Id.Data
+
+	svc := conn.Ec2(a.Region.Data)
+	ctx := context.Background()
+	endpoints := []interface{}{}
+	filterKeyVal := "vpc-id"
+	nextToken := aws.String("no_token_to_start_with")
+	params := &ec2.DescribeVpcEndpointsInput{Filters: []vpctypes.Filter{{Name: &filterKeyVal, Values: []string{vpc}}}}
+	for nextToken != nil {
+		endpointsRes, err := svc.DescribeVpcEndpoints(ctx, params)
+		if err != nil {
+			return nil, err
+		}
+		nextToken = endpointsRes.NextToken
+		if endpointsRes.NextToken != nil {
+			params.NextToken = nextToken
+		}
+
+		for _, endpoint := range endpointsRes.VpcEndpoints {
+			var subnetIds []interface{}
+			for _, subnet := range endpoint.SubnetIds {
+				subnetIds = append(subnetIds, subnet)
+			}
+			mqlEndpoint, err := CreateResource(a.MqlRuntime, "aws.vpc.endpoint",
+				map[string]*llx.RawData{
+					"id":             llx.StringData(*endpoint.VpcEndpointId),
```

Comment:
i haven't tested this one yet, but same comment as the other pr applies - if you can please add in the arn and have that be returned by the id function. if the arn isn't returned on the response, you can assemble it (there are a bunch of examples of this in the aws_ec2 resources  file